src/Game/Player.h

```diff
@@ -35,7 +35,7 @@ class Player final : public Component
     NON_SERIALIZED
     float additional_ship_speed = 0.2f * 0.005f;
     NON_SERIALIZED
-    float turn_speed = 15.0f;
+    float turn_speed = 0.0f;
```

Comment:
One would think that since starting `turn_speed` is equal to `0.0f`, ships would not turn at all, unless at the beginning of the game we set the starting values for these variables from the corresponding curves. However, we don't set these values at the beginning from the curves, but ships still turn seemingly correctly.

This is because `Ship`s have their own members like `turn_speed`, `range`, etc. which have default values **different** than those here. This leads to a weird situations, where ex. currently the default value for `range` in `Ship.h` is `0.5f`, but the value from the first point in the ships_range_curve is `0.4f`. In this case upgrading a lighthouse might actually ***decrease*** the range.

Now that I think about this, I don't really see a point for the values in `Ship.h`, since they should always have the same values as the variables in `Player.h` (which is currently a singleton, so retrieving the values is as easy as with Ship variables). Unless I forgot something that we talked about. I would probably delete those `Ship.h` static members and retrieve values directly from the Player. I would also set these values in Player at the start from the curves in LevelController.

Reply:
>This first category must definitely be stored in the ship because ships may have different maximum speeds at the same time during the game.

I'm aware of that. I was only talking about **`static`** variables (so the second category).

>Due to potential tests, it would be more convenient, for example, from the position of an inspector to change the value for only one ship.

Well, since these are static variables, changing them inside an editor of one ship, **will actually change the value for all ships**. But at the same time you also have variables with the same names in `Player.h`. Changing them either via code or editor, wouldn't actually do anything... until the lighthouse is upgraded. All of this is pretty confusing and bug prone.

>If in the future we would like to somehow define the differences between ships, for example larger ships would have smaller turn speed.

Not sure how storing these `static` variables like `turn_speed` in `Ship.h` rather than `Player.h` would help in that. I'm pretty sure implementing this is not possible in either of these solutions in a nice way. This is probably a "future us" problem.

>When we get to the point of designing the tutorial, it may turn out that the ships will be created by us, not by the ship spawner, and will be serialized. Then setting these values will be useful.

Static variables are not serialized, technically they could but I don't really see why. Even if they were serialized, all ships would still share the same value.

>Perhaps the least important, but it seems to me that operating the turn_speed variable in the code is more convenient than using Player::get_instance()->turn_speed.

We can implement shorter methods like `Player::get_turn_speed()`, `Player::turn_speed()`, or something like that.

Reply:
I forgot that all these variables are static so that I can change them for all ships. In such a case, what I am writing actually makes little sense. I'll fix it in the next commit. This is perfect because I was planning to remodel the ships' behavior to work according to a state machine in order to pass an artificial intelligence assignment.